src/code.sh

```
@@ -71,82 +83,73 @@ collect_rnaseq_metrics() {
+    fi
+
+    ## Setup 
+    dx-download-all-inputs
+
+    # Calculate 90% of memory size for java
+    mem_in_mb=$(head -n1 /proc/meminfo | awk '{print int($2*0.9/1024)}')
+
+    # Set java command with the calculated maximum memory usage
+    java="java -Xmx${mem_in_mb}m"
+
+    tar zxvf $fasta_index_path
```

Shellcheck (0.10.0)

[warning] fasta_index_path is referenced but not assigned.

(SC2154)